File: src/test/kotlin/com/boclips/event/service/testsupport/CollectionFactory.kt
package com.boclips.event.service.testsupport

import com.boclips.eventbus.domain.AgeRange
import com.boclips.eventbus.domain.Subject
import com.boclips.eventbus.domain.SubjectId
import com.boclips.eventbus.domain.collection.Collection
import com.boclips.eventbus.domain.collection.CollectionId
import com.boclips.eventbus.domain.user.UserId
import com.boclips.eventbus.domain.video.VideoId
import java.time.ZonedDateTime

object CollectionFactory {

    fun createCollection(
        id: String = "",
        title: String = "",
        description: String = "",
        subjects: Set<String> = emptySet(),
        ageRange: AgeRange = AgeRange(null, null),
        videoIds: List<String> = emptyList(),
        ownerId: String = "",
        createdTime: ZonedDateTime = ZonedDateTime.now(),
        updatedTime: ZonedDateTime = ZonedDateTime.now(),
        bookmarks: List<String> = emptyList(),
        isPublic: Boolean = true
    ): Collection {
        return Collection.builder()
            .id(CollectionId(id))
            .createdAt(createdTime)
            .updatedAt(updatedTime)
            .title(title)
            .description(description)
            .subjects(subjects.map { Subject(SubjectId("$it-id"), it) })
            .ageRange(ageRange)
            .videosIds(videoIds.map { VideoId(it) })
            .ownerId(UserId(ownerId))
            .bookmarks(bookmarks.map { UserId(it) })
            .isPublic(isPublic)
            .build()
    }
}

File: event-aggregator/src/test/scala/com/boclips/event/aggregator/testsupport/testfactories/CollectionFactory.scala
package com.boclips.event.aggregator.testsupport.testfactories

import java.time.ZonedDateTime

import com.boclips.event.aggregator.domain.model.collections.{Collection, CollectionId}
import com.boclips.event.aggregator.domain.model.users.UserId
import com.boclips.event.aggregator.domain.model.videos.VideoId
import com.boclips.event.aggregator.domain.model.{collections, _}

object CollectionFactory {

  def createCollection(
                        id: String = "123",
                        title: String = "title",
                        description: String = "description",
                        subjects: List[String] = List(),
                        ageRange: AgeRange = AgeRange(None, None),
                        videoIds: List[String] = List(),
                        ownerId: String = "owner",
                        bookmarks: List[String] = List(),
                        createdTime: ZonedDateTime = ZonedDateTime.now(),
                        updatedTime: ZonedDateTime = ZonedDateTime.now(),
                        deleted: Boolean = false,
                        public: Boolean = true,
                        promoted: Boolean = false
                      ): Collection =
    collections.Collection(
      id = CollectionId(id),
      title = title,
      description = description,
      subjects = subjects,
      ageRange = ageRange,
      videoIds = videoIds.map(VideoId),
      ownerId = UserId(ownerId),
      bookmarks = bookmarks.map(UserId),
      createdTime = createdTime,
      updatedTime = updatedTime,
      public = public,
      deleted = deleted,
      promoted = promoted,
    )
}
